tests/proxy/reliable_topic_test.py

```diff
@@ -0,0 +1,377 @@
+        item = reliable_topic.ringbuffer.read_one(seq).result()
+        num = self.client.serialization_service.to_object(item.payload)
+        self.assertEqual(num, 11)
+
+    def test_error(self):
+        reliable_topic = self.client.get_reliable_topic("error").blocking()
+        for i in range(10):
+            reliable_topic.publish(i+1)
+
+        with self.assertRaises(TopicOverflowError):
+            reliable_topic.publish(11)
+
+    def test_blocking(self):
+        reliable_topic = self.client.get_reliable_topic("blocking").blocking()
+        for i in range(10):
+            reliable_topic.publish(i+1)
+
+        before = datetime.utcnow()
+        reliable_topic.publish(11)
+        time_diff = datetime.utcnow() - before
+
+        seq = reliable_topic.ringbuffer.tail_sequence().result()
+        item = reliable_topic.ringbuffer.read_one(seq).result()
+        num = self.client.serialization_service.to_object(item.payload)
+        self.assertEqual(num, 11)
+        if time_diff.seconds <= 2:
+            self.fail("expected at least 2 seconds delay got %s" % time_diff.seconds)
+
+    def test_stale(self):
+        collector = event_collector()
+        self.reliable_topic = self.client.get_reliable_topic("stale").blocking()
+        reliable_listener = TestReliableMessageListenerLossTolerant(collector)
+        self.registration_id = self.reliable_topic.add_listener(reliable_listener)
+
+        items = self.generate_items(20)
+        self.reliable_topic.ringbuffer.add_all(items, overflow_policy=OVERFLOW_POLICY_OVERWRITE)
+
+        def assert_event():
+            self.assertEqual(len(collector.events), 10)
+            event = collector.events[9]
+            self.assertEqual(event.message, 20)
+
+        self.assertTrueEventually(assert_event, 5)
+
+    def test_distributed_object_destroyed(self):
+        config = ClientConfig()
+        config.network_config.connection_attempt_limit = 10
+        config.set_property(ClientProperties.INVOCATION_TIMEOUT_SECONDS.name, 10)
+        config.set_property("hazelcast.serialization.input.returns.bytearray", True)
+
+
+        client_two = hazelcast.HazelcastClient(self.configure_client(config))
+        # TODO: shutdown
+
+        collector = event_collector()
+        self.reliable_topic = client_two.get_reliable_topic("x")
+        reliable_listener = TestReliableMessageListenerLossTolerant(collector)
+        self.registration_id = self.reliable_topic.add_listener(reliable_listener)
+
+        self.rc.shutdownCluster(self.cluster.id)
+        self.cluster = self.create_cluster(self.rc, self.configure_cluster())
+        self.cluster.start_member()
+
+        self.reliable_topic.publish("aa")
+
+        def assert_event():
+            self.assertEqual(len(collector.events), 1)
+            event = collector.events[0]
+            self.assertEqual(event.message, "aa")
+
+        self.assertTrueEventually(assert_event, 5)
```

Comment:
You can shutdown the client_two by putting this line in try block and performing shutdown in the finally block.